client/src/components/CreatePost/Form/TabForms.js

```diff
@@ -27,18 +27,31 @@ const ModalComponent = ({
   gtmTagPrefix,
 }) => {
   const { t } = useTranslation();
-  const tabs = [
-    {
-      key: "offer",
+  const tabs = [];
+  const typeOfHelp = {
+    "offer":{
+      key:"offer",
       title: t("post.givingHelp"),
       question: t("post.whatGiving"),
     },
-    {
-      key: "request",
+    "request":{
+      key:"request",
       title: t("post.gettingHelp"),
       question: t("post.whatNeed"),
-    },
-  ];
+    }
+  }
+
+  if (currentPost){
+    if (currentPost.objective){
```

Comment:
just for corner case when `currentPost` is not null but `currentPost.objective` is null, maybe combine them to be like:
```
if (currentPost?.objective)
```

Reply:
That's an <strong>excellent</strong> idea. Thanks for the tip! I pushed the branch.